### ISSUE TITLE
Add a new Cystoid Nest Mechanics.

### ISSUE THREAD

**Issue by Caaz** [Caaz/rimworld-typhon#124]
# Description
Cystoids can wind up lingering on the map. A quick fix right now will be having them leave the map after a while, but a more fun opportunity can happen here.

Cystoids should be able to join together to create new cystoid nests, or join existing nests. increasing the amount of cystoids that will be created when they burst.

# Art Needed
Potentially various new sizes of cystoid nests

**Comment by TheErrorInC0de** [comment]
![Cystoid Nest zeroG](https://user-images.githubusercontent.com/122708353/212798580-c7550dd3-df31-45ac-be14-7492c9c49f1f.PNG)
Redownloaded this game to assist finding references for you guys, this is an instance of a Cystoid next growing on a wall outside Talos-1, I am searching for an instance in a non-zero-gravity environment but I am not sure if one exists in game

**Comment by TheErrorInC0de** [comment]
Potentially better image of a cystoid nest:
![Cystoid Nest GUTS](https://user-images.githubusercontent.com/122708353/212802777-98cf179a-b6cf-42d1-8c15-839f9d8ba9cc.PNG)
Followed by the frames I've got of it exploding (Cant compress my video file enough for Github to be happy with me sending it so frames is all you get for now):
Immediately after bullet impact, brighter colors:
![Cystoid Nest Pop 1of3](https://user-images.githubusercontent.com/122708353/212802891-38be514f-1454-4574-8ff3-49cc37a30f2d.PNG)
Primary shockwave of the explosion:
![Cystoid Next Pop 2of3](https://user-images.githubusercontent.com/122708353/212802985-b46a2cdf-e792-43cd-9733-c7646d352474.PNG)
Secondary boom:
![Cystoid Nest Pop 3of3](https://user-images.githubusercontent.com/122708353/212803026-de8affe1-2964-4003-b65d-2b88ce5a4b3d.PNG)
And then it spawns cystoids themselves:
![Cysstoid Nest Pop Spawn](https://user-images.githubusercontent.com/122708353/212803120-9109daba-da72-46d5-a7f4-b3c0f6d7af32.PNG)

Thats what I've got for now on the cystoid matter, again will try to find a nest in a non-zero-gravity environment if I can.

**Comment by Caaz** [comment]
Nice, thank you! Yeah I think the non-zero-g environments have them on the ceiling and they generally look exactly like the zero-g ones, only problem in rimworld is roofs aren't visible and perspectives mean nothing. Might be interesting with some SoS2 integration to have them attach to walls though, if possible.

**Comment by TheErrorInC0de** [comment]
Well I did find instances of non-zero-g cystoid nests and you where right:
![Cystoid Nest In Gravity](https://user-images.githubusercontent.com/122708353/212989051-c323c03e-9e3c-48aa-ba55-ab4338178691.PNG)
They normally hang from the ceiling in gravity, however the smaller more bulbus ones are the only version of cystoid nests that stick to the ground in gravity influenced zones, which makes Polioguy's artwork accurate, its just missing the orange lines dotted about its surface.

**Comment by Caaz** [comment]
Good point. I'd love to spend some time code-wise and get those visual effects working in a glowey manner, every time I've tried adding to the sprite it doesn't feel right. I might take advantage of some overlay stuff when I come back around to cystoids.
